HLF Quest

DSGMods opened the pull request:
**- What is the purpose of the pull request?**
Adds a new quest to the Bat Caverns
**- Give a brief description of what you changed or added.**
BatCaverns, BatCavernsEncounterDialogue, WorldType, PlaceType, QuestLine, QuestTree modified the most. All edits are either additions or brand new files
Overview of quest flow:
-Player finds entrance to new map via random encounter
-Entrance is locked with password
-Player needs to find two halves of the password via two more random encounters
-Player can loot the new map at their leisure
-Looting the armory will start a subquest to get more firebombs from Roxy
-The final room contains a loredump and a combat encounter
-Leaving without reading the loredump fails the quest
-Leaving after reading the loredump completes it
-Speaking to Roxy and waiting 2 days will complete the subquest
**- Are any new graphical assets required?**
All required graphical assets have been included, already optimized via SVGO
**- Has this change been tested? If so, mention the version number that the test was based on.**
Tested on 0.3.9.9
**- So we have a better idea of who you are, what is your Discord Handle?**
dsg#4931

DSGMods committed:
Initial HLF mod commit

DSGMods committed:
Merge pull request #1 from Innoxia/dev

Accept hotfix

DSGMods committed:
Merge pull request #2 from DSGMods/dev

Accept hotfix into hlf_mod branch

DSGMods committed:
Quest fixes

DSGMods committed:
Merge pull request #3 from Innoxia/dev

0.3.9.2 Merge

DSGMods committed:
Merge pull request #4 from DSGMods/dev

Merge into HLF Mod

DSGMods committed:
Change encounter chance code to not increase the chance of nulls when player has completed or failed quest

DSGMods committed:
Merge pull request #6 from Innoxia/dev

Accept 0.3.9.3 changes

DSGMods committed:
Password and revisiting work

DSGMods committed:
0.3.9.3 compatibility patch

DSGMods committed:
Merge branch 'hlf_mod' into dev

DSGMods committed:
Merge pull request #7 from DSGMods/dev

Accept 0.3.9.3 changes

DSGMods committed:
0.3.9.3 fixes

DSGMods committed:
Quest expansion update

DSGMods committed:
Prep for PR, add set bonuses, and misc fixes

DSGMods committed:
Prep for PR, add set bonuses, and misc fixes

CognitiveMist commented on nbactions.xml:
```diff
@@ -0,0 +1,17 @@
+                <exec.executable>java</exec.executable>
+            </properties>
+        </action>
+    </actions>
```
I don't think this file should be committed

DSGMods replied:
You are right, for some reason this file was included in the commit despite being in the ignore list. I've let Innoxia know to delete.

CognitiveMist commented on pom.xml:
```diff
@@ -6,7 +6,7 @@
 	<packaging>jar</packaging>
 	<version>1</version>
-	<name>lilithsthrone</name>
+	<name>lilithsthrone_dsgmodenv</name>
```
I don't think this change should be committed

DSGMods replied:
I don't either, I'm not sure why the project got renamed. I let Innoxia know to revert.

CognitiveMist commented on res/clothing/dsg/hlf_equip/rbandolier.xml:
```diff
@@ -0,0 +1,184 @@
+
+		<!-- These tags determine where in the world your clothing can be found, and what special attributes your clothing should have. Possible tags are found here: https://github.com/Innoxia/liliths-throne-public/blob/master/src/com/lilithsthrone/game/inventory/ItemTag.java -->
+		<itemTags> 
+			<tag>SOLD_BY_NYAN</tag>
```
Would Nyan sell this uniform?

DSGMods replied:
No, I must have missed this tag, I'll delete it.

DSGMods committed:
Remove SOLD_BY_NYAN and SOLD_BY_VICKY tags from all new items

DSGMods committed:
Merge pull request #8 from Innoxia/dev

v0.3.9.7 merge

DSGMods committed:
Merge branch 'hlf_mod' into dev

DSGMods committed:
Merge pull request #9 from DSGMods/dev

v0.3.9.7 merge

DSGMods commented:
v0.3.9.7 conflicts resolved

DSGMods committed:
Merge pull request #10 from Innoxia/dev

Merge 0.3.9.8 changes

DSGMods committed:
Fixed many typos
Select clothing items and all weapons now use the contraband system
Fixed a bug where it was possible to bypass the password check at the cave door if the first part of the password was found

DSGMods committed:
Merge pull request #11 from Innoxia/dev

Accept 0.3.9.9 changes

DSGMods committed:
Check commit

DSGMods committed:
Standardize all stroke colors on new items to RGB black instead of CMYK black.

DSGMods committed:
Merge pull request #12 from Innoxia/dev

Update after previous commit

DSGMods committed:
Revert changes to line endings introduced in commit ab0f6e42996463673bf0a3c7d4ca1995f4eb00a6

DSGMods committed:
Update teargas shotgun to latest weapon standard, add critical hit.
Fix petrolbomb unequip.

DSGMods committed:
Updated journal entries to better reflect recent lore

Stadler76 commented:
AceXP commented [on 3 Sep 2020](https://github.com/Innoxia/liliths-throne-public/pull/1398#issuecomment-686404895):
> This PR has some conflicts at the moment, so maybe (if you are up to it) you can try to resolve those.

Again has unresolved conflicts.

DSGMods commented:
> 
> 
> AceXP commented [on 3 Sep 2020](https://github.com/Innoxia/liliths-throne-public/pull/1398#issuecomment-686404895):
> 
> > This PR has some conflicts at the moment, so maybe (if you are up to it) you can try to resolve those.
> 
> Again has unresolved conflicts.

![image](https://user-images.githubusercontent.com/50091859/106064490-946b1d80-60bf-11eb-84be-759541f17a2b.png)

DSGMods committed:
Initial insane survivor work

DSGMods committed:
Revert "Initial insane survivor work"

This reverts commit db5c8040e37638a7bafc3805d57fed4adc601088.

DSGMods committed:
Add dirty version of HLF tunic
Re-add insane survivor because git sucks

DSGMods committed:
Initial HLF Quest Insane Survivor encounter work, also fixed some typos in hitTexts

DSGMods committed:
Insane survivor attack work complete.
Added new job perk, updated quest level

DSGMods committed:
Added new itemTag for Roxy
Roxy can now sell arcane firebombs after the player talks to her

AceXP commented on nbactions.xml:
```diff
@@ -1,17 +1,32 @@
-            </properties>
-        </action>
-    </actions>
+<?xml version="1.0" encoding="UTF-8"?>
```
I don't think this file should be in the change list?

DSGMods replied:
This was added to the gitignore, not sure how it got here. I will delete it but I have no idea if the next commit will add it back.

AceXP commented on res/txt/places/submission/gamblingDen/roxysShop.xml:
```diff
@@ -232,6 +232,44 @@
+	
+	<htmlContent tag="FIREBOMBS"><![CDATA[
+	<p>
+		You think back to the Arcane Firebombs you got from the mysterious cave. You don't have a way to get any more and you doubt any honest shop would want to carry something so obviously illegal. An idea starts forming in your back of your mind...
```
should be 'in the back of your mind'

AceXP commented on res/txt/encounters/submission/rebelBase/InsaneSurvivorAttack.xml:
```diff
@@ -0,0 +1,103 @@
+
+	<htmlContent tag="ATTACK"><![CDATA[
+	<p>
+		As you enter the room, you notice a ghostly figure crouched in the corner with [npc.her] back turned to you, mumbling something inaudibly and seemingly unaware of your presence. While you deliberate whether or not to approach [npc.her], you must have made some sort of noise since [npc.she] suddenly jumps up and turns to face you. A half-eaten glowing mushroom falls from [npc.her] emaciated hand ...
```
Should be 'approach [npc.himher]'

AceXP commented on src/com/lilithsthrone/game/character/npc/submission/RebelBaseInsaneSurvivor.java:
```diff
@@ -0,0 +1,249 @@
+            this.setEssenceCount(500);
+            setDescription(UtilText.parse(this, "While apparently mostly human, [npc.name] has evidently spent a long time underground and has lost [npc.her] mind."));
+            
+            CharacterUtils.generateItemsInInventory(this);
```
I get this error on this line: Non-static method 'generateItemsInInventory(com.lilithsthrone.game.character.npc.NPC)' cannot be referenced from a static context
Should be changed to this I think: Main.game.getCharacterUtils().generateItemsInInventory(this);

DSGMods replied:
Please do not build on any other version of LT other than 3.9.9

AceXP commented on src/com/lilithsthrone/game/character/npc/submission/RebelBaseInsaneSurvivor.java:
```diff
@@ -0,0 +1,249 @@
+            this.setStartingCombatMoves();
+            
+            // Nobody wants to see the gyrations of a living crack zombie
+            this.unequipMove("tease");
```
This doesn't seem to work, tease moves are still generated.
To fix it, add this to override some methods:
```
    @Override
    public void setStartingCombatMoves() {
        this.clearEquippedMoves();
        this.equipMove("strike");
        this.equipMove("twin-strike");
    }

    @Override
    public float getMoveWeight(AbstractCombatMove move, List<GameCharacter> enemies, List<GameCharacter> allies) {
        // Death or Glory!
        if (move.getType() == CombatMoveType.TEASE || move.getType() == CombatMoveType.DEFEND) {
            return 0;
        } else {
            return super.getMoveWeight(move, enemies, allies);
        }
    }
```

DSGMods replied:
Fixed

DSGMods replied:
Although getMoveWeight does not accept AbstractCombatMove parameters, changed to CombatMove (as in Silence.java)

AceXP commented on src/com/lilithsthrone/game/dialogue/npcDialogue/submission/RebelBaseInsaneSurvivorDialogue.java:
```diff
@@ -0,0 +1,218 @@
+                return new Response("Offer money", "Your empty pockets don't give you the means to try this.", null);
+            }
+            else if (index == 3) {
+                return new Response("Offer money", "[npc.She] probably isn't after money it might be worth a try.", INSANE_SURVIVOR_BRIBE_ATTEMPT);
```
Missing 'but', maybe change it so it's a little different from the 'Talk' option?
'While [npc.she] probably isn't after money it could be worth trying it.'

DSGMods replied:
Fixed

DSGMods committed:
Various bug and typo fixes from @AceXP

DSGMods committed:
Delete nbactions.xml

DSGMods committed:
Delete pom.xml

DSGMods committed:
Added Elle reactions to the player wearing HLF sets

DSGMods committed:
Fixed various code formatting issues
Changed dealing with Roxy to a quest to assist less attentive players
Removed superfluous SOLD_BY_ROXY tag as it wasn't being utilized anymore

DSGMods committed:
Quest logic fixes

Innoxia commented:
221 days later...
Thank you so much for this! The quest is great, not to mention how amazing the new clothing and weapons are!
